src/core/Xtarget.ml

```diff
@@ -1,16 +1,56 @@
-(* eXtended target.
+(* Yoann Padioleau, Cooper Pierce
  *
- * This type is mostly used in the engine, to pass around extra information
- * associated to each target.
+ * Copyright (c) Semgrep Inc.
  *
- * related: Input_to_core.target, which is what is passed
- * to semgrep-core via -target.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** eXtended target.
+
+   This type is mostly used in the engine to pass around extra information
+   (e.g., contents, the AST) associated with each {{!Target.regular}target}.
+
+   See also {!Input_to_core_t.target}, which is what is passed to
+   [semgrep-core] via [-target].
  *)
 
 type t = {
-  file : Fpath.t;
-  xlang : Xlang.t;
+  path : Target.path;
+  xlang : Xlang.t;  (** The analyzer to use when scanning this target. *)
   lazy_content : string lazy_t;
-  (* This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
   lazy_ast_and_errors : (AST_generic.program * Tok.location list) lazy_t;
+      (** This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
 }
+
+let parse_file parser (analyzer : Xlang.t) path =
```

Comment:
now that this type has a few functions, please add a Xtarget.mli

Reply:
Done in d57830ad536d2800a7e4c53159e6c156191eae83.